redigomock.go

```diff
@@ -182,12 +182,15 @@ func (c *Conn) do(commandName string, args ...interface{}) (reply interface{}, e
 	c.stats[cmd.hash()]++
 	c.statsMut.Unlock()
 
+	cmd.Called = true
 	if len(cmd.Responses) == 0 {
 		return nil, nil
 	}
 
 	response := cmd.Responses[0]
-	cmd.Responses = cmd.Responses[1:]
+	if len(cmd.Responses) > 1 {
```

Comment:
I didn't meant to just check if it nill, it's actually a convenient patch so you will get the last response value if the response already reach the end. I thought it will have backward compatibility but not if reaching the end and nil value is expected.